yt/frontends/nc4_cm1/data_structures.py

```diff
@@ -0,0 +1,233 @@
+            )
+            self.domain_right_edge = np.array(
+                [xh.max(), yh.max(), zh.max()], dtype="float64"
+            )
+
+            # loop over the variable names in the netCDF file, record only those on the
+            # "zh","yh","xh" grid.
+            varnames = []
+            for key in ds.variables.keys():
+                vardims = ds.variables[key].dimensions
+                if all(x in vardims for x in ["time", "zh", "yh", "xh"]) is True:
+                    varnames.append(key)
+            self.parameters["variable_names"] = varnames
+            self.parameters["lofs_version"] = ds.cm1_lofs_version
+            self.parameters["is_uniform"] = ds.uniform_mesh
+            self.current_time = ds.variables["time"][:][0]
+
+            # record the dimension metadata
+            dim_info = OrderedDict()
+            for dim, meta in ds.dimensions.items():
+                dim_info[dim] = meta.size
+            self.parameters["dimensions"] = dim_info
+
+        self.parameters["coords"] = coords
+
+        self.dimensionality = 3
+        self.domain_dimensions = np.array(dims, dtype="int64")
+        self.periodicity = (False, False, False)
+        self.parameters["time"] = self.current_time
+
+        # Set cosmological information to zero for non-cosmological.
+        self.cosmological_simulation = 0.0
+        self.current_redshift = 0.0
+        self.omega_lambda = 0.0
+        self.omega_matter = 0.0
+        self.hubble_constant = 0.0
+
+    @classmethod
+    def _is_valid(self, *args, **kwargs):
+        # This accepts a filename or a set of arguments and returns True or
+        # False depending on if the file is of the type requested.
+
+        warn_netcdf(args[0])
+        try:
+            from netCDF4 import Dataset
+
+            with Dataset(args[0], "r", keepweakref=True) as ds:
+                is_cm1_lofs = hasattr(ds, "cm1_lofs_version")
+                is_cm1 = hasattr(ds, "cm1 version")
+                # ensure coordinates of each variable array exists in the dataset
+                coords = ds.dimensions  # get the dataset wide coordinates
+                failed_vars = []  # list of failed variables
+                for var in ds.variables:  # iterate over the variables
+                    vcoords = ds[var].dimensions  # get the dimensions for the variable
+                    ncoords = len(vcoords)  # number of coordinates in variable
+                    coordspassed = 0  # number of coordinates that pass for a variable
+                    for vc in vcoords:  # iterate over the coordinates for the variable
+                        if vc in coords:
+                            # variable coordinate and global coordinate are same
+                            coordspassed += 1
+                    if coordspassed != ncoords:
+                        failed_vars.append(var)
+
+                if len(failed_vars) > 0:
```

Comment:
python for the win !
```suggestion
                if failed_vars:
```